src/GitVersionCore/GitRepoMetadataProvider.cs

```diff
@@ -77,7 +77,7 @@ public IEnumerable<Branch> GetBranchesContainingCommit(Commit commit, IList<Bran
                 // TODO: It looks wasteful looping through the branches twice. Can't these loops be merged somehow? @asbjornu
                 foreach (var branch in branches)
                 {
```

Comment:
This is the change I talked about in the PR. What the original dev probably meant was to use `onlyTrackedBranches || !ibranch.IsTracking`. I used the longer version `(onlyTrackedBranches && branch.IsTracking) || !onlyTrackedBranches` as its more verbose and makes sure this bug doesn't happen :)